modules/http4s/src/main/scala/pureconfig/module/http4s/package.scala

```diff
@@ -1,16 +1,40 @@
 package pureconfig.module
 
+import com.comcast.ip4s.{Hostname, Port}
 import org.http4s.Uri
 
 import pureconfig.error.CannotConvert
 import pureconfig.{ConfigReader, ConfigWriter}
 
 package object http4s {
 
+  implicit val uriWriter: ConfigWriter[Uri] =
+    ConfigWriter[String].contramap(_.renderString)
+
   implicit val uriReader: ConfigReader[Uri] =
     ConfigReader.fromString(str =>
       Uri.fromString(str).fold(err => Left(CannotConvert(str, "Uri", err.sanitized)), uri => Right(uri))
     )
 
-  implicit val uriWriter: ConfigWriter[Uri] = ConfigWriter[String].contramap(_.renderString)
+  implicit val hostnameWriter: ConfigWriter[Hostname] =
+    ConfigWriter[String].contramap(_.toString)
+
+  implicit val hostnameReader: ConfigReader[Hostname] =
+    ConfigReader.fromString(str =>
+      Hostname.fromString(str) match {
+        case Some(hostname) => Right(hostname)
+        case None => Left(CannotConvert(str, "Hostname", "Invalid hostname"))
+      }
+    )
+
+  implicit val portWriter: ConfigWriter[Port] =
+    ConfigWriter[Int].contramap(_.value)
+
+  implicit val portReader: ConfigReader[Port] =
+    ConfigReader.intConfigReader.emap(str =>
```

Comment:
You shouldn't rely on `ConfigReader.intConfigReader` here, as `intConfigReader` is meant to be consumed as an implicit. You should use `ConfigReader[Int]` instead, similarly to how you summon a writer above.

```suggestion
    ConfigReader[Int].emap(str =>
```

Reply:
Sorry, good to know, just updated the PR

Reply:
Done